runtime/interpreter/interpreter.go

```diff
@@ -3026,6 +3000,58 @@ var stringFunction = func() Value {
 		),
 	)
 
+	addMember(
+		sema.StringTypeFromUtf8FunctionName,
+		NewUnmeteredHostFunctionValue(
+			func(invocation Invocation) Value {
+				argument, ok := invocation.Arguments[0].(*ArrayValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
+
+				inter := invocation.Interpreter
+
+				// meter byte count, not utf-8 codepoints
+				memoryUsage := common.NewArrayExpressionMemoryUsage(
+					argument.Count(),
+				)
```

Comment:
The byte array was already metered when it was created. What we're doing here is metering the new string we are going to return, so this should be a `NewStringMemoryUsage`, rather than an array expression. Also, so that we don't incorrectly meter a string when we return `nil` from this function; put the usage calculation down in the `if` statement that actually returns the string.